libraries/heft-config-file/src/ConfigurationFile.ts

```diff
@@ -147,21 +161,55 @@ export class ConfigurationFile<TConfigurationFile> {
     string,
     IConfigurationFileCacheEntry<TConfigurationFile>
   > = new Map<string, IConfigurationFileCacheEntry<TConfigurationFile>>();
+  private readonly _fileExistsCache: Map<string, boolean> = new Map<string, boolean>();
   private readonly _packageJsonLookup: PackageJsonLookup = new PackageJsonLookup();
 
   public constructor(options: IConfigurationFileOptions<TConfigurationFile>) {
+    this._projectRelativeFilePath = options.projectRelativeFilePath;
     this._schemaPath = options.jsonSchemaPath;
-    this._jsonPathMetadata = options?.jsonPathMetadata || {};
-    this._propertyInheritanceTypes = options?.propertyInheritanceTypes || {};
+    this._jsonPathMetadata = options.jsonPathMetadata || {};
+    this._propertyInheritanceTypes = options.propertyInheritanceTypes || {};
+    this._rigsEnabled = !options.disableRigs;
   }
 
-  public async loadConfigurationFileAsync(configurationFilePath: string): Promise<TConfigurationFile> {
-    return await this._loadConfigurationFileInnerAsync(
-      nodeJsPath.resolve(configurationFilePath),
-      new Set<string>()
+  public async loadConfigurationFileForProjectAsync(projectPath: string): Promise<TConfigurationFile> {
+    const projectConfigurationFilePath: string = this._getConfigurationFilePathForProject(projectPath);
+    return await this._loadConfigurationFileInnerWithCacheAsync(
+      projectConfigurationFilePath,
+      new Set<string>(),
+      this._rigsEnabled ? projectPath : undefined
     );
   }
 
+  /**
+   * This function is identical to {@link ConfigurationFile.loadConfigurationFileForProjectAsync}, except
+   * that a preliminary file existence check is performed and this function returns `undefined` if the
+   * configuration file doesn't exist.
+   */
+  public async tryLoadConfigurationFileForProjectAsync(
```

Comment:
I wonder if we should eventually move this into `rig-package`.

For example, we were chatting about adding a setting like this:

**rig.json**
```js
{
  "$schema": "https://developer.microsoft.com/json-schemas/rig-package/rig.schema.json",

  "rigPackageName": "heft-minimal-rig-test",
  "disabledFilePaths": [
    // The rig enables API Extractor, but we want to disable it locally for this project
    "config/api-extractor.json"
  ]
}
```

Any such logic would need to be reimplemented by each tool that loads files.  Heft is adding some extra logic to cache lookups, but maybe `rig-package` could support an `IRigPackageHost` type interface that allows the caller to optionally provide a custom implementation of `existsAsync()`

Reply:
That'll be a major PITA without access to node-core-library, especially now that we've added logging here.